harness/determined/pytorch/_trainer.py

```diff
@@ -0,0 +1,211 @@
+import contextlib
+import logging
+import random
+from typing import Dict, Optional, Union, cast
+
+import torch
+import torch.distributed as dist
+
+import determined as det
+from determined import core, horovod, pytorch
+from determined.horovod import hvd
+from determined.profiler import DummyProfilerAgent, ProfilerAgent
+
+
+class Trainer:
+    def __init__(self, trial: pytorch.PyTorchTrial, context: pytorch.PyTorchTrialContext):
+        self._trial = trial
+        self._context = context
+        self._cluster_info = det.get_cluster_info()
```

Comment:
why bother saving this to `self`?  It's already a singleton, and you already need an assert everytime you use it.